httpx/dispatch/asgi.py

```diff
@@ -121,13 +120,7 @@ def __init__(
                 await response_body.mark_as_done()
                 response_started_or_failed.set()
 
-        # Really we'd like to push all `asyncio` logic into concurrency.py,
-        # with a standardized interface, so that we can support other event
-        # loop implementations, such as Trio and Curio.
-        # That's a bit fiddly here, so we're not yet supporting using a custom
-        # `ConcurrencyBackend` with the `Client(app=asgi_app)` case.
-        loop = asyncio.get_event_loop()
-        app_task = loop.create_task(run_app())
+        background = await self.backend.background_manager(run_app).__aenter__()
```

Comment:
It's worth us just being aware that this is a *really* weak design point for us right now.
We're essentially saying "hey this background task is context-managed 👍, haha not really 😝".

(Not a critque on the pull request, just raising awareness around an existing problem point that this is touching on.)